server/src/main/java/org/elasticsearch/snapshots/InternalSnapshotsInfoService.java

```diff
@@ -65,9 +66,11 @@ public class InternalSnapshotsInfoService implements ClusterStateListener, Snaps
     private final Supplier<RerouteService> rerouteService;
 
     /** contains the snapshot shards for which the size is known **/
+    // volatile for the unlocked access in numberOfKnownSnapshotShardSizes()
+    // map itself is immutable
     private volatile Map<SnapshotShard, Long> knownSnapshotShards;
 
-    private volatile boolean isMaster;
```

Comment:
While I think this is correct, the boolean is only modified inside a sync block, this change is outside the scope of the PR. We should make a separate PR to address different issues and tag the appropriate folks to review.

Reply:
#90087 for after this PR is merged